src/org/edx/hadoop/input/ManifestTextInputFormat.java

```diff
@@ -13,20 +9,36 @@
 import org.apache.hadoop.fs.Path;
 import org.apache.hadoop.mapred.JobConf;
 import org.apache.hadoop.mapred.KeyValueTextInputFormat;
-import org.apache.commons.logging.LogFactory;
 
 
 public class ManifestTextInputFormat extends KeyValueTextInputFormat {
+    public static int RETRY_LIMIT = 10;
+    public static long REPORTING_INTERVAL = 1000l * 60 * 5;
 
     protected FileStatus[] listStatus(JobConf job) throws IOException {
         FileStatus[] manifests = super.listStatus(job);
         List<FileStatus> paths = new ArrayList<FileStatus>();
-        for(int i = 0; i < manifests.length; i++) {
-            List<Path> globPaths = this.readManifest(manifests[i].getPath(), job);
-            for (Path globPath : globPaths) {
+        long lastReportTime = System.currentTimeMillis();
+
+        for (FileStatus manifest : manifests) {
+            List<Path> globPaths = this.readManifest(manifest.getPath(), job);
+
+            for(int i = 0; i < globPaths.size(); i++) {
+                Path globPath = globPaths.get(i);
+
                 paths.addAll(this.expandPath(globPath, job));
+
+                if ((System.currentTimeMillis() - lastReportTime) >= REPORTING_INTERVAL) {
+                    lastReportTime = System.currentTimeMillis();
+
+                    if (globPath != null)
+                        LOG.info("Fetched " + i + " of " + globPaths.size() + " paths, most recently: " + globPath.toUri());
+                    else
```

Comment:
Does this happen?  That is, is globPath sometimes null?   Because if it is, we should not be adding it to paths on line 29.  The call to expandPath (defined below) does not protect against passing in a null value -- it will fail at line 64.